### PR TITLE
FAT-1697: Implement e-2-e automation of test case C350591

### PR DESCRIPTION
**Pull request by Javokhir12**
## Refs
- https://foliotest.testrail.io/index.php?/cases/view/350591

### CODE REVIEW

**Comment by Javokhir12** [comment]
Hi @dmitriyfolio, @MaksimTerekhov, @TetianaParanich. Can you review it? We need to close it today. Thanks.

File: cypress/integration/data-import/match-on-vendor-reference-number.spec.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
+/* eslint-disable cypress/no-unnecessary-waiting */
```
Comment (dmitriyfolio) [inline cypress/integration/data-import/match-on-vendor-reference-number.spec.js:1]: why do you need such disabling?

File: cypress/fixtures/matchOnVendorReferenceNumber.mrc [hunk 2]
```diff
@@ -0,0 +1 @@
...
  aDubuque, IA :bMcGraw-Hill,cc2010.
  axxv, 518 p. :bcol. ill., maps ;c28 cm.
  aDott's name appears first on the earlier edition.
  aIncludes index.
  aIncludes bibliographical references and index.
 0aHistorical geologyvTextbooks.
07aHistorische Geologie.2swd
1 aDott, Robert H.,d1929-
  aKU/CC/DI/MbQE28.3c.D68 2010
  a1bszt97166227181891
```
[the remark this answers is not in the document]
Comment (Javokhir12) [reply cypress/fixtures/matchOnVendorReferenceNumber.mrc:1]: @dmitriyfolio Unfortunately, no. In testcase, it is specified that we need to have specific file with couple hard-coded values.

[the remark this answers is not in the document]
Comment (dmitriyfolio) [reply cypress/fixtures/matchOnVendorReferenceNumber.mrc:1]: can you take existing marc file and edit it, add few special properties which required to this test?  As in our current regular case when user imports marc file, change some properties and export updated file. After this - continue to work through the test

File: cypress/integration/data-import/match-on-vendor-reference-number.spec.js (hunk 1 above)
Comment (Javokhir12) [reply cypress/integration/data-import/match-on-vendor-reference-number.spec.js:1]: It is just to disable lint warnings.

File: cypress/integration/data-import/match-on-vendor-reference-number.spec.js [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+  it('C350591 Match on VRN and update related Instance, Holdings, Item', { tags: [TestTypes.smoke] }, () => {
+    // create order
+    cy.visit(TopMenu.ordersPath);
+    Orders.createOrder(order);
+    MatchOnVRN.verifyCreatedOrder(order);
+
+    // add and verify po line
+    Orders.createPOLineViaActions();
+    MatchOnVRN.fillPOLineInfo();
```
[the remark this answers is not in the document]
Comment (Javokhir12) [reply cypress/integration/data-import/match-on-vendor-reference-number.spec.js:99]: if I am not mistaken it is saying that Contributor and Qualifier fields are optional so we don't have to fill it.

File: cypress/integration/data-import/match-on-vendor-reference-number.spec.js [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+    MatchOnVRN.goBackToPO();
+    MatchOnVRN.openOrder();
+    MatchOnVRN.verifyOrderStatus();
+
+    // create field mapping profiles
+    cy.visit(SettingsMenu.mappingProfilePath);
+    MatchOnVRN.creatMappingProfilesForInstance(instanceMappingProfileName)
```
Comment (dmitriyfolio) [inline cypress/integration/data-import/match-on-vendor-reference-number.spec.js:106]: step in TestCase and implementation should be fully crossed. Think to redesign related steps or update your implementation

File: cypress/integration/data-import/match-on-vendor-reference-number.spec.js [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+    MatchOnVRN.verifyInstanceStatusNotUpdated();
+    MatchOnVRN.clickOnUpdatedHotlink();
+    MatchOnVRN.verifyInstanceUpdated();
+    MatchOnVRN.verifyHoldingsUpdated();
+    MatchOnVRN.verifyItemUpdated();
+    MatchOnVRN.verifyMARCBibSource();
+
+    MatchOnVRN.deleteItem();
+    MatchOnVRN.deleteHoldings();
+    MatchOnVRN.deletePOLine();
```
[the remark this answers is not in the document]
Comment (Javokhir12) [reply cypress/integration/data-import/match-on-vendor-reference-number.spec.js:146]: Correct. After test finished running, this step deletes created `po line`.

File: cypress/integration/data-import/match-on-vendor-reference-number.spec.js (hunk 4 above)
Comment (Javokhir12) [reply cypress/integration/data-import/match-on-vendor-reference-number.spec.js:106]: could you clarify what step is not implemented? I checked test case and my implementation has the same steps as in the testcase

**Review by MaksimTerekhov — APPROVED** [review]

File: cypress/integration/data-import/match-on-vendor-reference-number.spec.js [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+import Helper from '../../support/fragments/finance/financeHelper';
+import TopMenu from '../../support/fragments/topMenu';
+import Logs from '../../support/fragments/data_import/logs/logs';
+import MatchOnVRN from '../../support/fragments/data_import/matchOnVRN';
+import FileManager from '../../support/utils/fileManager';
+import SettingsMenu from '../../support/fragments/settingsMenu';
+import Orders from '../../support/fragments/orders/orders';
+import getRandomPostfix from '../../support/utils/stringTools';
+import permissions from '../../support/dictionary/permissions';
+import users from '../../support/fragments/users/users';
```
Comment (penfriend) [inline cypress/integration/data-import/match-on-vendor-reference-number.spec.js:13]: Please import fragments starting from UpperCase letter

**Review by penfriend — APPROVED** [review]

**Pull request merged**